test/runtests.jl

```diff
@@ -182,22 +200,36 @@ ac2 = convert(ARGB32, c)
 @test convert(ARGB32, 0x01020304).color == 0x01020304
 ac3 = convert(RGBA, ac)
 @test convert(RGB24, ac3) == c
+ac4 = AGray32(.2,.8)
+@test alpha(ac4) == .8
+@test gray(ac4) == .2
 
 for C in subtypes(AbstractRGB)
     rgb = convert(C, c)
     C == RGB24 && continue
+    @test ccolor(Gray24, C) == Gray24
+    @test ccolor(AGray32, C) == AGray32
     @test convert(AbstractRGB, c) == c
     @test convert(AbstractRGB{Float64}, rgb) === convert(C{Float64}, c)
     argb = convert(alphacolor(C), ac)
+    rgba = convert(coloralpha(C), ac)
     @test rgb.r == red(c)
     @test rgb.g == green(c)
     @test rgb.b == blue(c)
     @test argb.alpha == alpha(ac)
     @test argb.r == red(ac)
     @test argb.g == green(ac)
     @test argb.b == blue(ac)
+    @test rgba.alpha == alpha(ac)
+    @test rgba.r == red(ac)
+    @test rgba.g == green(ac)
+    @test rgba.b == blue(ac)
 end
 
+@test convert(Float64, Gray(.3)) == .3
```

Comment:
You can combine this and the next line with `===` (3 equals signs), which requires that the left and right sides be indistinguishable (including type).